Orange/widgets/data/owcolor.py

```diff
@@ -350,6 +350,7 @@ def set_data(self, data):
                                         self._create_proxies(domain.metas))
             self.openContext(data)
             self.data = data.transform(domain)
+            self.data.domain = domain
```

Comment:
> ... and risk @janezd changing it back to `data.transform` the next time he sees this code? 

@janezd change to `data.transform` didn't break anything (`transform` and `to_table` behave the same way). Removing of the extra `self.data.domain = domain` is the change that broke things.

> If data.transform is syntactic sugar it has to have the same behaviour as the thing that this sugar is coating.

Agree. Hence, if we are going to fix this in the core we need to do it in `from_table`. And while I'm all for changing `to_table` to compare domains with `is` instead of `==`,  I'm afraid this might cause other issues all over the place. As it has revealed some problems in domain's `__contains__` which I'm not sure how to solve. Any help appreciated 🙈